models/DataObject/Concrete.php

```diff
@@ -788,6 +797,14 @@ public function __wakeup()
         if (property_exists($this, 'localizedfields') && $this->localizedfields instanceof Localizedfield) {
             $this->localizedfields->setObject($this, false);
         }
+
+        // renew object reference to other object aware fields
+        foreach($this->__objectAwareFields as $objectAwareField => $exists) {
+            $brickContainer = $this->$objectAwareField;
+            if($brickContainer instanceof ObjectAwareFieldInterface) {
+                $brickContainer->setObject($this);
+            }
```

Comment:
```suggestion
            if(isset($this->$objectAwareField) && $this->$objectAwareField instanceof ObjectAwareFieldInterface) {
                $this->$objectAwareField->setObject($this);
            }
```
I'd add `isset($this->$objectAwareField)` here, just in case the class definition changed in the meantime and a field definition was removed.

Reply:
but `$brickContainer instanceof ObjectAwareFieldInterface` doesn't evaluate to true, if `$this->$objectAwareField` is not set, right?

Reply:
But it would throw an `Warning: Undefined property: ....` error.

Reply:
Just verified that: 
```php
        $image = Asset\Image::getById(360);

        // works
        if(isset($image->sdfdsgdfgfghfgh) && $image->sdfdsgdfgfghfgh instanceof Asset) {
            $foo = 'bar';
        }

        // throws Warning: Undefined property: Pimcore\Model\Asset\Image::$sdfdsgdfgfghfgh
        if($image->sdfdsgdfgfghfgh instanceof Asset) {
            $foo = 'bar';
        }
```